go/arrow/csv/common.go

```diff
@@ -22,12 +22,88 @@ import (
 	"errors"
 	"fmt"
 	"github.com/apache/arrow/go/arrow"
+	"github.com/apache/arrow/go/arrow/memory"
 )
 
 var (
 	ErrMismatchFields = errors.New("arrow/csv: number of records mismatch")
 )
 
+// Option configures a CSV reader/writer.
+type Option func(config)
+type config interface{}
+
+// WithComma specifies the fields separation character used while parsing CSV files.
+func WithComma(c rune) Option {
+	return func(cfg config) {
+		switch cfg := cfg.(type) {
+		case *Reader:
+			cfg.r.Comma = c
+		case *Writer:
+			cfg.w.Comma = c
+		default:
+			panic(fmt.Errorf("arrow/csv: unknown config type %T", cfg))
+		}
+	}
+}
+
+// WithComment specifies the comment character used while parsing CSV files.
+func WithComment(c rune) Option {
+	return func(cfg config) {
+		switch cfg := cfg.(type) {
+		case *Reader:
+			cfg.r.Comment = c
+		default:
+			panic(fmt.Errorf("arrow/csv: unknown config type %T", cfg))
+		}
+	}
+}
+
+// WithAllocator specifies the Arrow memory allocator used while building records.
+func WithAllocator(mem memory.Allocator) Option {
+	return func(cfg config) {
+		switch cfg := cfg.(type) {
+		case *Reader:
+			cfg.mem = mem
+		default:
+			panic(fmt.Errorf("arrow/csv: unknown config type %T", cfg))
+		}
+	}
+}
+
+// WithChunk specifies the chunk size used while parsing CSV files.
+//
+// If n is zero or 1, no chunking will take place and the reader will create
+// one record per row.
+// If n is greater than 1, chunks of n rows will be read.
+// If n is negative, the reader will load the whole CSV file into memory and
+// create one big record with all the rows.
+func WithChunk(n int) Option {
+	return func(cfg config) {
+		switch cfg := cfg.(type) {
+		case *Reader:
+			cfg.chunk = n
+		default:
+			panic(fmt.Errorf("arrow/csv: unknown config type %T", cfg))
+		}
+	}
+}
+
+// WithCRLF specifies the line terminator used while writing CSV files.
+// if useCRLF is True, \r\n is used as the line terminator/
```

Comment:
there's a stray `/` (after terminator).
is that on purpose?

also: `s/if useCRLF is True/If useCRLF is true/`

perhaps:
`s|line terminator/|line terminator otherwise \n is used.|`